zcash_client_sqlite/src/wallet/scanning.rs

```diff
@@ -0,0 +1,1098 @@
+            Joined::One(ScanRange::from_parts(
+                left.block_range().start..right.block_range().end,
+                left.priority(),
+            ))
+        } else {
+            Joined::Two(left, right)
+        }
+    } else {
+        // there is a gap that will need to be filled
+        let gap = ScanRange::from_parts(
+            left.block_range().end..right.block_range().start,
+            ScanPriority::Historic,
+        );
+
+        match join_nonoverlapping(left, gap) {
+            Joined::One(left) => join_nonoverlapping(left, right),
+            Joined::Two(left, gap) => match join_nonoverlapping(gap, right) {
+                Joined::One(right) => Joined::Two(left, right),
+                Joined::Two(gap, right) => Joined::Three(left, gap, right),
+                _ => unreachable!(),
+            },
+            _ => unreachable!(),
+        }
+    }
+}
+
+fn insert(current: ScanRange, to_insert: ScanRange) -> Joined {
+    fn join_overlapping(left: ScanRange, right: ScanRange, insert: Insert) -> Joined {
+        assert!(
+            left.block_range().start <= right.block_range().start
+                && left.block_range().end > right.block_range().start
+        );
+
+        // recompute the range dominance based upon the queue entry priorities
+        let dominance = match insert {
+            Insert::Left => dominance(&right.priority(), &left.priority(), insert),
+            Insert::Right => dominance(&left.priority(), &right.priority(), insert),
+        };
+
+        match dominance {
+            Dominance::Left => {
+                if let Some(right) = right.truncate_start(left.block_range().end) {
+                    Joined::Two(left, right)
+                } else {
+                    Joined::One(left)
+                }
+            }
+            Dominance::Equal => Joined::One(ScanRange::from_parts(
+                left.block_range().start..max(left.block_range().end, right.block_range().end),
+                left.priority(),
+            )),
+            Dominance::Right => match (
+                left.truncate_end(right.block_range().start),
+                left.truncate_start(right.block_range().end),
+            ) {
+                (Some(left), Some(end)) => Joined::Three(left, right, end),
+                (Some(left), None) => Joined::Two(left, right),
+                (None, Some(end)) => Joined::Two(right, end),
+                (None, None) => Joined::One(right),
+            },
+        }
+    }
+
+    use RangeOrdering::*;
+    match RangeOrdering::cmp(to_insert.block_range(), current.block_range()) {
+        LeftFirstDisjoint => join_nonoverlapping(to_insert, current),
+        LeftFirstOverlap | RightContained => join_overlapping(to_insert, current, Insert::Left),
+        Equal => Joined::One(ScanRange::from_parts(
+            to_insert.block_range().clone(),
+            update_priority(current.priority(), to_insert.priority()),
```

Comment:
```suggestion
            match dominance(current.priority(), to_insert.priority(), Insert::Right) {
                Dominance::Left | Dominance::Equal => current.priority(),
                Dominance::Right => to_insert.priority(),
            }
```
and delete `update_priority`, as suggested in https://github.com/zcash/librustzcash/pull/872/files#r1266470051 .